modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/DataSet.js

```diff
@@ -200,7 +200,7 @@ const DataSet = ({
 	}
 
 	useEffect(() => {
-		const itemsAreInjected = !apiURL && itemsProp?.length !== items.length;
+		const itemsAreInjected = !apiURL && itemsProp?.length > 0;
```

Comment:
I tested your suggestion here and this `useEffect` needs this validation, otherwise the tables that are using the `apiURL` will be broken because the table items will always be the `itemsProps` which in this case will always be empty. But we can simplify the logic this way, what do you think?

```tsx

useEffect(() => {	
        if (!apiURL) {
	        updateDataSetItems({items: itemsProp});
        }
}, [apiURL, itemsProp]);

```

Reply:
Nice catch! I tested this, and it doesn't break, but only because of the other `useEffect` that invokes `updateDataSetItems` as well.

Maybe a cleaner safeguard would be to check if `itemsProp` was set, something like 
```js
useEffect(() => {	
        if (itemsProp !== undefined) {
	        updateDataSetItems({items: itemsProp});
        }
}, [itemsProp]);
```
We still want it to update with an empty array.

For this check to work, we need to remove `itemsProp` from `defaultProps`, are remove default props assignemt ` = [],`.

With this, in theory, we would be able to have component both controlled any uncontrolled at the same time.

Reply:
Awesome @markocikos, works for both cases! I'm sending the changes Thanks!